Refactor code

fairyinabottle4 opened the pull request:
All references to AddressBook and Person are changed to
StudentBook and Student respectively

fairyinabottle4 committed:
Refactor code

All references to AddressBook and Person are changed to
StudentBook and Student respectively

fairyinabottle4 committed:
Merge DG changes

fairyinabottle4 committed:
Update DG with Student changes

fairyinabottle4 committed:
Update DG with Student changes

dannn-tran commented on docs/DeveloperGuide.md:
```diff
@@ -133,69 +133,90 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
-#### Actual Implementation
+#### Proposed Implementation
 
-The `Add` Student mechanism is implemented in the `AddCommand` class and facilitated by the following classes:
-* `Command`. `AddCommand` extends `Command` and overrides the `execute` method, which adds a `Person` to the `Model`. 
-* `AddCommandParser`. It implements the `Parser` interface, which is used by all commands to read user input. `AddCommandParser` also checks the parameters of `AddCommand` and make sures that the input adheres to the specified format. 
+The proposed undo/redo mechanism is facilitated by `VersionedAddressBook`. It extends `StudentBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`. Additionally, it implements the following operations:
```
Have you not merged with the latest master branch, to reflect the new DG?

fairyinabottle4 committed:
Merge branch 'master' into update-student

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/diagrams/ModelClassDiagram.puml
#	docs/diagrams/StorageClassDiagram.puml
#	docs/images/ModelClassDiagram.png
#	docs/images/StorageClassDiagram.png
#	src/main/java/seedu/address/model/util/SampleDataUtil.java
#	src/test/java/seedu/address/testutil/PersonBuilder.java

fairyinabottle4 committed:
Refactor code

Update DG to reflect the new class names.
Package name also changed and refactored.

fairyinabottle4 committed:
Refactor code

fairyinabottle4 committed:
Run Checkstyle

fairyinabottle4 committed:
Merge branch 'master' into update-student

# Conflicts:
#	src/main/java/seedu/student/logic/parser/FindCommandParser.java

fairyinabottle4 committed:
Refactor FindCommandParser

yienyoong reviewed (approved):
Thank you so much for doing this part of our tp.
